src/lib.rs

```diff
@@ -1,3 +1,117 @@
+//! A library to help convert from a unicode string and attributes to a list of glyphs and
+//! positions.
+//!
+//! Converting unicode strings to glyphs is a complex process, requiring knowledge of
+//!  - which fonts to use (and choosing an appropriate font given a size, script, typeface, weight,
+//!    etc.),
+//!  - a description of how to render a font (e.g. a truetype font file)
+//!  - the space available for drawing and what to do if it's not enough,
+//!  - and information on the user's locale.
+//!
+//! # Font terms
+//!
+//! The terminology of printing can be confusing. Partly this is because printing has been so
+//! important for so long, and so has historical baggage, but it is also because unifying the
+//! written methods of many different cultures requires a complicated system with many parts.
+//! Unfortunately, a consequence of this is that most of the terms used in typesetting have
+//! different meanings to different people. The unicode standard is useful in that it precisely
+//! defines the terms it uses, but these meanings aren't necessarily the same as those that are in
+//! common usage.  Below are some important terms in font rendering and the meanings some ascribe
+//! to them:
+//!
+//! This is a work in progress and will be wrong in places (for now).
+//!
+//!  - *typography* - The study of drawing text to paper, screen, or some other medium. Drawing
```

Comment:
I think this is a very useful guide, and very useful to have in the module-level docs. I can see expanding it in time. Below I have a bunch of comments, but we don't have to revise it completely, especially thanks to the disclaimer above.

I think *generally* it would be useful for most of these to have pointers to an authoritative source. Sometimes that will be a particular place in the Unicode spec (including the annexes). Other times, Wikipedia is reasonable.

Reply:
Could I do this in a future pass? It seems daunting to do it now.